cmd/thanos/store.go

```diff
@@ -50,11 +50,23 @@ func registerStore(m map[string]setupFunc, app *kingpin.Application, name string
 	blockSyncConcurrency := cmd.Flag("block-sync-concurrency", "Number of goroutines to use when syncing blocks from object storage.").
 		Default("20").Int()
 
+	minTime := store.TimeOrDuration(cmd.Flag("min-time", "Start of time range limit to serve. Option can be a constant time in RFC3339 format or time duration relative to current time, such as -1.5d or 2h45m. Valid duration units are ms, s, m, h, d, w, y.").
+		Default("0000-01-01T00:00:00Z"))
```

Comment:
hm.. I think the earliest date might be `-9999-12-31T23:59:59Z` but not 100% sure ;p to be strict

Reply:
it doesn't time.Parse:

```
unexpected error: parsing time "-9999-12-31T23:59:59Z" as "2006-01-02T15:04:05Z07:00": cannot parse "-9999-12-31T23:59:59Z" as "2006"
```

```
func TestTime(t *testing.T) {
	ti, err := time.Parse(time.RFC3339, "-9999-12-31T23:59:59Z")
	testutil.Ok(t, err)

	fmt.Printf("time.Unix: %v, prom.timestamp: %v\n", ti.Unix(), timestamp.FromTime(ti))
}

```

Reply:
;p